department-of-veterans-affairs/caseflow
# HunJerBAH APPEALS-13375: Letter Task Cancel Actions

Resolves [APPEALS-13364](https://vajira.max.gov/browse/APPEALS-13364), [APPEALS-13368](https://vajira.max.gov/browse/APPEALS-13368), and [APPEALS-13375](https://vajira.max.gov/browse/APPEALS-13375)

### Description
Created cancel actions for the `SendInitialNotificationLetterTask`, `PostSendInitialNotificationLetterHoldingTask`, and `SendFinalNotificationLetterTask` and updated cancel task modals to reflect the acceptance criteria. 

### Acceptance Criteria

**SendInitialNotificationLetterTask AC**
- [ ] `SendInitialNotificationLetterTask` cancel action cancels the task
- [ ] `SendInitialNotificationLetterTask` cancel action modal matches AC of [APPEALS-13364](https://vajira.max.gov/browse/APPEALS-13364)
- [ ] `SendInitialNotificationLetterTask` shows in the Case Timeline as `cancelled` after the task is cancelled. 
- [ ] Actions are only accessible by members of the COB team.

**PostSendInitialNotificationLetterHoldingTask AC**
- [ ] `PostSendInitialNotificationLetterHoldingTask` cancel action cancels the task
- [ ] `PostSendInitialNotificationLetterHoldingTask` cancel action modal matches AC of [APPEALS-13368](https://vajira.max.gov/browse/APPEALS-13368)
- [ ] `PostSendInitialNotificationLetterHoldingTask` shows in the Case Timeline as `cancelled` after the task is cancelled. 
- [ ] Actions are only accessible by members of the COB team.


**SendFinalNotificationLetterTask AC**
- [ ] `SendFinalNotificationLetterTask` cancel action cancels the task
- [ ] `SendFinalNotificationLetterTask` cancel action modal matches AC of [APPEALS-13375](https://vajira.max.gov/browse/APPEALS-13375)
- [ ] `SendFinalNotificationLetterTask` shows in the Case Timeline as `cancelled` after the task is cancelled.
- [ ] Actions are only accessible by members of the COB team.

### Testing Plan
**Note: as of this PR, the `PostSendInitialNotificationLetterHoldingTask` and  `SendFinalNotificationLetterTask` do not have a testing plan since the logic to create these tasks has not been completed. When the logic is completed, the testing plan will be similar to the `SendInitialNotificationLetterTask` testing plan below.**


1. Go to Caseflow login and sign in as `BVADWISE`
2. Navigate to Mail Intake 
3. Select the first option from the dropdown and click the button to continue
![image](https://user-images.githubusercontent.com/99915461/216086553-ade4b4c3-c238-46bd-bc7b-367efcf88ba2.png)

4. Enter a veteran file number. I used 484190004. Click the search icon to continue
![image](https://user-images.githubusercontent.com/99915461/216086704-e42827cc-a3c0-4480-9526-4b0eb1d5e8fa.png)

5. Fill out the form and select **direct review** for the review option type. Click the button at the bottom to continue
![image](https://user-images.githubusercontent.com/99915461/216087208-8ce1a1df-0c71-4aac-bb1a-949161fade72.png)

6. On the add issues page, click the **add issues** button. Select a benefit type of **insurance** and a **contested claim** issue. Click **add this issue** to add the issue and **establish appeal** to complete the intake.
![image](https://user-images.githubusercontent.com/99915461/216087480-5a1f1673-8a85-4794-aa85-9d0c2ceeb96e.png)

7. The intake should be completed. Copy the veteran id and click **search cases**.
![image](https://user-images.githubusercontent.com/99915461/216087966-95e51461-ba07-4d07-8a94-3b45c1086219.png)

8. Enter the veteran file number into the search bar and select **search**. The issue you are looking for will have a purple **CC** icon next to it. Click the docket number link to get to the veteran's case details page.
![image](https://user-images.githubusercontent.com/99915461/216088534-086299db-f78a-47c5-b82e-9bd4e30a897f.png)

9. In the currently active tasks, the **Send Initial Notification Letter** should show and be assigned to **COB**. No actions should display.
![image](https://user-images.githubusercontent.com/99915461/216088794-e4dff6c7-23cd-407d-81b0-98c85bd37b25.png)

10. To see the action list, you will need to sign in as a COB member. Click the user name in the right corner and **switch users** from the dropdown. On the user login page, sign in as a COB user
![image](https://user-images.githubusercontent.com/99915461/216089168-db6b84e6-8920-4875-a4e7-cd5b64ae4c06.png)
![image](https://user-images.githubusercontent.com/99915461/216089332-e2835428-9220-4ec9-87b2-f974682ca1ac.png)

11. Navigate to Queue and click **search cases** in the top right corner. 
12. Enter the veteran file number and repeat step 8 to navigate to the case details screen. The Actions dropdown should display now
![image](https://user-images.githubusercontent.com/99915461/216089744-dfe3874f-ab0b-4e71-bac1-757986eb0cd9.png)

13. Click the actions dropdown and select **cancel task**
![image](https://user-images.githubusercontent.com/99915461/216089870-559b901f-8a20-469f-8e13-e72f9aaced98.png)

14. Confirm the **Cancel Task** modal matches the AC for APPEALS-13364. 
![image](https://user-images.githubusercontent.com/99915461/216090050-86ac19f7-c2a3-4156-816f-68726fc6bedc.png)

15. Click **submit**. You should navigate to the user's queue and be greeted with a success banner. 
![image](https://user-images.githubusercontent.com/99915461/216090533-57ed6f72-827d-48ba-a166-577556fd7d38.png)

16. Repeat steps 11 and 12 to navigate back to the veteran's case detail screen.
17. The active task should be the distribution task with no action list. The Initial task should show as cancelled in the case timeline at the bottom of the page. If you click on **view task instructions**, you should see the instructions, who cancelled the task, and who the task was assigned to.
![image](https://user-images.githubusercontent.com/99915461/216091372-57c6b146-c92c-4043-b30e-0a89b8125f27.png)
![image](https://user-images.githubusercontent.com/99915461/216091457-b4d38541-b652-47fb-addf-bbc1f4e9310a.png)
![image](https://user-images.githubusercontent.com/99915461/216091877-43373927-bcc5-4834-8d8a-43a15b38a2d7.png)
